src/integrations/watchers/indexable-ancestor-watcher.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Author watcher to save the meta data to an Indexable.
+ *
+ * @package Yoast\YoastSEO\Watchers
+ */
+
+namespace Yoast\WP\SEO\Integrations\Watchers;
+
+use Yoast\WP\SEO\Builders\Indexable_Builder;
+use Yoast\WP\SEO\Builders\Indexable_Hierarchy_Builder;
+use Yoast\WP\SEO\Conditionals\Migrations_Conditional;
+use Yoast\WP\SEO\Helpers\Indexable_Helper;
+use Yoast\WP\SEO\Integrations\Integration_Interface;
+use Yoast\WP\SEO\Models\Indexable;
+use Yoast\WP\SEO\Repositories\Indexable_Hierarchy_Repository;
+use Yoast\WP\SEO\Repositories\Indexable_Repository;
+
+class Indexable_Ancestor_Watcher implements Integration_Interface {
+
+	/**
+	 * Represents the indexable repository.
+	 *
+	 * @var Indexable_Repository
+	 */
+	protected $indexable_repository;
+
+	/**
+	 * Represents the indexable hierarchy builder.
+	 *
+	 * @var Indexable_Hierarchy_Builder
+	 */
+	protected $indexable_hierarchy_builder;
+
+	/**
+	 * Represents the indexable helper.
+	 *
+	 * @var Indexable_Helper
+	 */
+	protected $indexable_helper;
+
+	/**
+	 * Sets the needed dependencies.
+	 *
+	 * @param Indexable_Repository        $indexable_repository        The indexable repository.
+	 * @param Indexable_Hierarchy_Builder $indexable_hierarchy_builder The hierarchy builder.
+	 * @param Indexable_Helper            $indexable_helper            The indexable helper.
+	 */
+	public function __construct(
+		Indexable_Repository $indexable_repository,
+		Indexable_Hierarchy_Builder $indexable_hierarchy_builder,
+		Indexable_Helper $indexable_helper
+	) {
+		$this->indexable_repository        = $indexable_repository;
+		$this->indexable_hierarchy_builder = $indexable_hierarchy_builder;
+		$this->indexable_helper            = $indexable_helper;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function register_hooks() {
+		\add_action( 'wpseo_save_indexable', [ $this, 'clear_ancestors' ], \PHP_INT_MAX, 2 );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function get_conditionals() {
+		return [ Migrations_Conditional::class ];
+	}
+
+	/**
+	 * Removes the ancestors when the permalink for the given indexable has been changed.
+	 *
+	 * @param Indexable $indexable        The indexable.
+	 * @param Indexable $indexable_before The old indexable.
+	 *
+	 * @return bool When clearing has been done.
+	 */
+	public function clear_ancestors( $indexable, $indexable_before ) {
```

Comment:
I would rename this function, because the name doesn't really reflect the functionality. Suggestion: `reset_children`.